projects/noloan/app/src/main/java/hasadna/noloan/InboxRecyclerAdapter.java

```diff
@@ -23,11 +31,47 @@
     extends RecyclerView.Adapter<InboxRecyclerAdapter.RecyclerViewHolder> {
   private static final String TAG = "InboxRecyclerAdapter";
 
-  public InboxRecyclerAdapter() {
-    if (SmsMessages.get().getInboxMessages().size() == 0) {
+  ArrayList<SmsMessage> messages;
+
+  public InboxRecyclerAdapter(ArrayList<SmsMessage> inbox) {
+    messages = inbox;
+    if (messages.size() == 0) {
       SmsMessage noMessage = SmsMessage.newBuilder().setSender("אין הודעות").build();
-      SmsMessages.get().getInboxMessages().add(noMessage);
+      messages.add(noMessage);
     }
+    Handler handler = new Handler(Looper.getMainLooper());
+
+    // If new spams added to the DB - Check if the spams exists in user's inbox, if so update the
+    // counter of suggesters
+    DbMessages.getInstance().addMessagesListener(new DbMessages.MessagesListener() {
+
+      @Override
+      public void messageAdded(SmsMessage smsMessage) {
+        // Check if user has this message in the inbox
+        int i = messages.indexOf(smsMessage);
```

Comment:
Search inbox messages by Body & Sender..
`SmsMessage` objects coming from the DB, won't probably never be equal to the ones in the inbox<SmsMessage> list.